services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_utils.py

```diff
@@ -0,0 +1,221 @@
+                        dynamic_sidecar_endpoint
+                    )
+                ]
+
+                # When enabled no longer uploads state via nodeports
+                # It uses rclone mounted volumes for this task.
+                if not app_settings.DIRECTOR_V2_DEV_FEATURE_R_CLONE_MOUNTS:
+                    tasks.append(
+                        dynamic_sidecar_client.save_service_state(
+                            dynamic_sidecar_endpoint
+                        )
+                    )
+
+                await logged_gather(*tasks, max_concurrency=2)
+                scheduler_data.dynamic_sidecar.was_data_saved_when_closing = True
+
+                logger.info("Ports data pushed by dynamic-sidecar")
+            except (BaseClientHTTPError, TaskClientResultError) as e:
+                logger.error(
+                    (
+                        "Could not contact dynamic-sidecar to save service "
+                        "state or upload outputs %s\n%s"
+                    ),
+                    scheduler_data.service_name,
+                    f"{e}",
+                )
+                # ensure dynamic-sidecar does not get removed
+                # user data can be manually saved and manual
+                # cleanup of the dynamic-sidecar is required
+
+                scheduler_data.dynamic_sidecar.wait_for_manual_intervention_after_error = (
+                    True
+                )
+                raise e
+
+    if dynamic_sidecar_settings.DYNAMIC_SIDECAR_DOCKER_NODE_RESOURCE_LIMITS_ENABLED:
+        node_rights_manager = NodeRightsManager.instance(app)
+        assert scheduler_data.dynamic_sidecar.docker_node_id  # nosec
+        try:
+            async with node_rights_manager.acquire(
+                scheduler_data.dynamic_sidecar.docker_node_id,
+                resource_name=RESOURCE_STATE_AND_INPUTS,
+            ):
+                await _remove_containers_save_state_and_outputs()
+        except NodeRightsAcquireError:
+            # Next observation cycle, the service will try again
+            logger.debug(
+                "Skip saving service state for %s. Docker node %s is busy. Will try later.",
+                scheduler_data.node_uuid,
+                scheduler_data.dynamic_sidecar.docker_node_id,
+            )
+            return
+    else:
+        await _remove_containers_save_state_and_outputs()
+
+    # remove the 2 services
+    await remove_dynamic_sidecar_stack(
+        node_uuid=scheduler_data.node_uuid,
+        dynamic_sidecar_settings=dynamic_sidecar_settings,
+    )
+    # remove network
+    await remove_dynamic_sidecar_network(scheduler_data.dynamic_sidecar_network_name)
+
+    # NOTE: when adding volume removal, check that:
+    # `scheduler_data.dynamic_sidecar.was_data_saved_when_closing` was
+    # set to True before removing data otherwise do nothing
+
+    logger.debug(
+        "Removed dynamic-sidecar created services for '%s'",
```

Comment:
this log is incorrect I think. at this point you have everything removed but some networks